### ISSUE TITLE
Add option to store settings in APEX folder

### ISSUE THREAD

**Issue by tomfrancart** [exporl/apex#264]
Currently, APEX settings such as calibration, or SPIN configurations are stored using QSettings. On Linux they go to a user-specific config-file and on windows they go to the registry.
We'd like an option to store the settings in a file in the APEX installation folder instead (so they become system-wide instead of user-specific, and also specific to that particular APEX version on the system).
Please add an option to apexconfig.xml to en/disable this behaviour.

The current use-case is to share profiles of the SPIN runner across users, the relevant code for that is in src/lib/apexspin/spinusersettingsdatabase.cpp
Apart from that "git grep QSettings" might be helpful

See also http://doc.qt.io/qt-5/qsettings.html#Format-enum

**Comment by linusdm** [comment]
QSettings is flexible in the way it looks for a value and has built-in support for differentiating between user and system-wide settings (http://doc.qt.io/qt-5/qsettings.html#Scope-enum). Would it suffice to make the scope for the settings configurable (system-wide vs per-user)? This is an easy change.
If the requirement is to have per-user versus per-application-installation settings, this requires more work, and will require that we save the settings in a file on windows installations, instead of in the registry.

**Comment by tomfrancart** [comment]
Not really: the problem is that for system wide settings, apex would need to run as administrator, and it quickly becomes a mess when you both have user specific and system wide settings. We already have this feature for calibration (cf the calibrationadmin tool, which allows to make a calibration setup system-wide).
I therefore suggest to add a option to make settings apex-installation-specific (the user will then need write permission to the apex installation folder, perhaps keep in mind that we might add an option later on to set the settings path in apexconfig.xml). This will involve setting QSettings such that settings are saved to file, probably comes down to one call to a method of QSettings and some magic to get the paths right ;)

**Comment by linusdm** [comment]
Yes, you're right, I wouldn't use the fallback mechanism for settings. It should be clear where settings are read and written.
Should we be concerned with the fact that existing apex installations would lose all their settings once the new version is used, because the information stored in the registry would not be seen anymore if we look at a user-specific folder instead?

I still see three scenarios with their pro's and con's:

1. All on filesystem
=> apex settings are configured user-specific @ filesystem in appdata per user
=> apex settings are configured installation-wide @ filesystem in apex installation folder
pro: settings can be truly installation-specific
pro: files are easier to manipulate
pro: all settings in the same format/place
con: existing installations would lose their user-settings (unless migrated somehow)

2. Toggle Scope, but keep everything in registry
=> apex settings are configured user-specific @ the windows registry
=> apex settings are configured system-wide @ the windows registry
pro: existing installations keep their settings
pro: no filesystem troubles on windows
pro: all settings in the same format/place
con: settings can't be installation-specific, but instead are system-wide
con: registry is not easily accessible and manipulated by users

3. Blend (user settings @ registry, installation-wide settings @ filesystem)
=> apex settings are configured user-specific @ the windows registry
=> apex settings are configured installation-wide @ filesystem in apex installation folder
pro: settings can be truly installation-specific
pro: files are easier to manipulate (installation-wide settings)
pro: existing installations keep their settings
con: settings are not in the same format/place
con: registry is not easily accessible and manipulated by users (user-specific settings)

I don't see how administrator rights are needed in any of the three scenario's, unless administrator rights are needed to write in the system-wide section of the windows registry?

Would it be ok to apply the solution to all the places where QSettings are used?
